src/libsync/configfile.h

```diff
@@ -227,6 +227,12 @@ class OWNCLOUDSYNC_EXPORT ConfigFile
     [[nodiscard]] bool launchOnSystemStartup() const;
     void setLaunchOnSystemStartup(const bool autostart);
 
+    [[nodiscard]] bool serverHasValidSubscription() const;
+    void setServerHasValidSubscription(const bool valid);
```

Comment:
no `const` in the method declaration
you are leaking an implementation detail in the interface (with POD data types, if you need `const`, it should be only when implementing the method as in the interface, it would not makes sense)